subprojects/core/src/main/java/org/gradle/api/internal/artifacts/configurations/RoleBasedConfigurationContainerInternal.java

```diff
@@ -145,44 +134,8 @@ public interface RoleBasedConfigurationContainerInternal extends ConfigurationCo
      *
      * If a configuration with this name already exists this method will <strong>overwrite</strong> its current usage to match what
      * would be set if the configuration needed to be created and emit an additional deprecation warning when doing this
-     * <strong>IFF</strong> {@code verifyPrexisting} is set to {@code true}.
+     * <strong>IFF</strong> {@code verifyPrexisting} is set to {@code true}.  Otherwise, it will throw a
+     * {@link GradleException}.
      */
     Configuration maybeCreateDependencyScopeLocked(String name, boolean verifyPrexisting);
```

Comment:
💭 It seems now this method is only used in one place -- `DefaultJvmFeature`, and for a very specific workaround for Kotlin. Does it make sense to just move this method into there? I would be suspicious of other code that ends up using this method

Reply:
It needs to call private methods in the `DefaultConfigurationContainer`, so it would be annoying to move.  We should definitely deprecate it in Gradle 9.x.  I'll add a note to it.